client/src/components/FormField/RegisteredMFAMethodListField/RegisteredMFAMethodListField.js

```diff
@@ -1,12 +1,42 @@
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
-
-import methodShape from '../../../types/registeredMethod';
+import { Button, Modal, ModalHeader, ModalBody } from 'reactstrap';
+import { inject } from 'lib/Injector'; // eslint-disable-line
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import methodShape from 'types/registeredMethod';
 import MethodListItem from './MethodListItem';
+import { showScreen, chooseMethod, setAvailableMethods } from 'state/mfaRegister/actions';
+import { SCREEN_CHOOSE_METHOD } from 'components/Register';
 
 const fallbacks = require('../../../../lang/src/en.json');
 
 class RegisteredMFAMethodListField extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      modalOpen: false,
+    };
+
+    props.onUpdateAvailableMethods(props.availableMethods);
```

Comment:
`availableMethods` or `onUpdateAvailableMethods`. `availableMethods` has a default so that's fine for fallback. I wasn't sure about prop definitions for those provided by Redux (`onUpdateAvailableMethods`). I can add all the Redux stuff as required props?

Reply:
Yeah may as well